### PR TITLE
Add SONiC WPA_SUPPLICANT Plugin

### PR DESCRIPTION
**Pull request by Pterosaur**
SONiC WPA_SUPPLICANT Plugin document can be found at : https://github.com/Azure/SONiC/blob/9edcf67d0e0095e9f831a94ddf4fbb4a267adf7b/doc/macsec/MACsec_hld.md#343-sonic-macsec-plugin

This PR depends on https://github.com/Azure/sonic-wpa-supplicant/pull/18

Signed-off-by: Ze Gan <ganze718@gmail.com>

### CODE REVIEW

**Review by caizhenghui — APPROVED** [review]
looks good to me.

File: src/drivers/driver_macsec_sonic.c [hunk 1]
```diff
@@ -0,0 +1,996 @@
...
+ * Returns: 0 on success, -1 on failure
+ */
+static int macsec_sonic_create_receive_sa(void *priv, struct receive_sa *sa)
+{
+    struct macsec_sonic_data *drv = priv;
+
+    char * key = CREATE_SA_KEY(drv->ifname, sa, APP_DB_SEPARATOR);
+    char * sak_id = create_binary_hex(&sa->pkey->key_identifier, sizeof(sa->pkey->key_identifier));
+    char * sak = create_binary_hex(sa->pkey->key, sa->pkey->key_len);
+    char * pn = create_buffer("%u", sa->next_pn);
```
Comment (samitabh) [inline src/drivers/driver_macsec_sonic.c:577]: sa->next_pn
should be
sa->lowest_pn

Similar changes below in the function.

File: src/drivers/driver_macsec_sonic.c [hunk 2]
```diff
@@ -0,0 +1,996 @@
...
+        sak_id,
+        sak);
+
+    // TODO
+    // SALT
+    const struct sonic_db_name_value_pair pairs[] = 
+    {
+        {"sak", sak},
+        {"auth_key", auth_key},
+        {"init_pn", pn},
```
Comment (samitabh) [inline src/drivers/driver_macsec_sonic.c:828]: init_pn --> this is actually next pn. You can change the name for clarity sake.

**Comment by Pterosaur** [comment]
This PR will be closed and moved to https://github.com/Azure/sonic-wpa-supplicant/pull/22 . Because the original version is based on branch `wpa/hostap_2_9`.

**Comment by Pterosaur** [comment]
@caizhenghui-juniper @samitabh 
Dear partners, If you have more comments, please move to the new PR: https://github.com/Azure/sonic-wpa-supplicant/pull/22
Thanks for your understanding。

**Pull request closed without merging**